api/apiclient_test.go

```diff
@@ -1087,6 +1089,36 @@ func (s *apiclientSuite) TestIsBrokenPingFailed(c *gc.C) {
 	c.Assert(conn.IsBroken(), jc.IsTrue)
 }
 
+func (s *apiclientSuite) TestLoginCapturesCLIArgs(c *gc.C) {
+	s.PatchValue(&os.Args, []string{"this", "is", "the test", "command"})
+
+	info := s.APIInfo(c)
+	conn := newRPCConnection()
+	conn.response = &params.LoginResult{
+		ControllerTag: "controller-" + s.ControllerConfig.ControllerUUID(),
+		ServerVersion: "2.3-rc2",
+	}
+	broken := make(chan struct{})
```

Comment:
A comment to explain this would be good

Reply:
Will add one - took me a while to figure it out, may as well store it for next time!